lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.ts

```diff
@@ -239,14 +247,41 @@ export class ContentMetadataComponent implements OnChanges, OnInit, OnDestroy {
         });
     }
 
-    /**
-     * Called after clicking save button. It confirms all changes done for metadata and hides both category and tag name controls.
-     * Before clicking on that button they are not saved.
-     */
-    saveChanges() {
+    onSaveGeneralInfoChanges(event?: MouseEvent) {
```

Comment:
Following functions are the same:
- onSaveGeneralInfoChanges
- onSaveTagsChanges
- onSaveCategoriesChanges
- onSaveGroupChanges - almost the same

Can we prepare reused function instead of duplicating?

Reply:
Denys splitted the switch cases into separate methods.
cc: @DenysVuika